spec/workers/company_domain_worker_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+RSpec.describe CompanyDomainWorker, sidekiq: true do
+  let!(:company) { create(:company, :worker_domain) }
+  let!(:worker) { CompanyDomainWorker.new }
+
+  before(:each) do
+    allow(worker).to receive(:link_open).and_return(response_html)
+
+    worker.perform(company.id)
+  end
+
+  describe 'with valid data' do
```

Comment:
use `describe` to separate components, and `context` to separate data conditions